src/libraries/Common/src/Interop/Windows/WinSock/Interop.WSADuplicateSocket.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Winsock
+    {
+        [StructLayout(LayoutKind.Sequential, CharSet=CharSet.Auto)]
+        internal struct WSAProtocolChain
+        {
+            internal int ChainLen;
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst=7)]
+            internal uint[] ChainEntries;
+        }
+
+        [StructLayout(LayoutKind.Sequential, CharSet=CharSet.Auto)]
+        internal struct WSAProtocolInfo
+        {
+            internal uint ServiceFlags1;
+            internal uint ServiceFlags2;
+            internal uint ServiceFlags3;
+            internal uint ServiceFlags4;
+            internal uint ProviderFlags;
+            internal Guid ProviderId;
+            internal uint CatalogEntryId;
+            internal WSAProtocolChain ProtocolChain;
+            internal int Version;
+            internal AddressFamily AddressFamily;
+            internal int MaxSockAddr;
+            internal int MinSockAddr;
+            internal SocketType SocketType;
+            internal ProtocolType ProtocolType;
+            internal int ProtocolMaxOffset;
+            internal int NetworkByteOrder;
+            internal int SecurityScheme;
+            internal uint MessageSize;
+            internal uint ProviderReserved;
+            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 256)]
+            internal string ProtocolName;
+
+            public static readonly int Size = Marshal.SizeOf(typeof(WSAProtocolInfo));
+        }
+
+        [DllImport(Interop.Libraries.Ws2_32, SetLastError = true)]
+        internal static extern unsafe SocketError WSADuplicateSocket(
```

Comment:
`WSADuplicateSocket` does not return a `SocketError`. It returns != 0 for error.

and then throw a new `SocketException` with default ctor or call `Marshal.GetLastWin32Error()` if you need the error.